beater/v2_handler.go

```diff
@@ -77,37 +77,48 @@ func (v *v2Handler) statusCode(sr *stream.Result) (int, *monitoring.Int) {
 	return highestCode, monitoringCt
 }
 
-func (v *v2Handler) sendResponse(logger *logp.Logger, w http.ResponseWriter, sr *stream.Result) {
+func (v *v2Handler) sendResponse(logger *logp.Logger, w http.ResponseWriter, r *http.Request, sr *stream.Result) {
 	statusCode, counter := v.statusCode(sr)
 	responseCounter.Inc()
 	counter.Inc()
 
-	w.WriteHeader(statusCode)
 	if statusCode != http.StatusAccepted {
 		responseErrors.Inc()
 		// this signals to the client that we're closing the connection
 		// but also signals to http.Server that it should close it:
 		// https://golang.org/src/net/http/server.go#L1254
 		w.Header().Add("Connection", "Close")
-		w.Header().Add("Content-Type", "application/json")
-		buf, err := json.Marshal(sr)
-		if err != nil {
-			logger.Errorw("error sending response", "error", err)
+		setContentType(w, r)
```

Comment:
just some ideas, but instead of `setContentType ` could we simplify by changing `sendJSON` and `sendPlain` to take `statusCode int` and then set the content type inside each one. Maybe we could also change `sendJSON` to take an `interface{}` instead and then use that both in v1 and v2. 

Then, instead of having a `sendStatus` in `common_handlers.go` which is only used in v1 (AFAICT), we could perhaps create one that we can use from both v1 and v2:

```
func sendResponse(w http.ResponseWriter, r *http.Request, statusCode int, response interface{}) {
    if acceptsJSON(r) {
         sendJSON(w, r, statusCode, response)
         return
    }
    sendPlain(w, r, statusCode, response)
}
```

Reply:
There are a couple of differences between v1 and v2 (in v2 Content-Type is only set on errors, counter for errors is increased in different places, the returned error messages are not the same). So I could align that and change some behaviour in v1, but I wasn't aiming for any change in v1, especially not within a bugfix. 
I think we should refactor that when we remove v1, but if you feel strongly about it I can come up with something.

Reply:
sure @simitt 👍